sqlx-core/src/any/row.rs

```diff
@@ -67,6 +72,30 @@ impl Row for AnyRow {
             AnyRowKind::Mssql(row) => row.try_get_raw(index).map(Into::into),
         }
     }
+
+    fn try_get<'r, T, I>(&'r self, index: I) -> Result<T, Error>
+    where
+        I: ColumnIndex<Self>,
+        T: Decode<'r, Self::Database> + Type<Self::Database>,
+    {
+        let value = self.try_get_raw(&index)?;
+
+        if !value.is_null() {
+            let ty = value.type_info();
+
+            if !ty.is_null() && !T::compatible(&ty) {
+                return Err(Error::ColumnDecode {
+                    index: format!("{:?}", index),
+                    source: mismatched_types::<T>(&ty),
+                });
+            }
+        }
+
+        T::decode(value).map_err(|source| Error::ColumnDecode {
+            index: format!("{:?}", index),
+            source,
+        })
```

Comment:
Does this current implementation optimize out `.type_info()` access? Or, can we simply do this instead?

```suggestion
        let value = self.try_get_raw(&index)?;
        let ty = value.type_info();

        if !value.is_null() && !ty.is_null() && !T::compatible(&ty) {
            Err(mismatched_types::<T>(&ty))
        } else {
            T::decode(value)
        }
        .map_err(|source| Error::ColumnDecode {
            index: format!("{:?}", index),
            source,
        })
```

Reply:
That's probably fine.